source/db/dffml_source_db/source.py

```diff
@@ -0,0 +1,143 @@
+import collections
+from typing import Type, AsyncIterator, List
+
+from dffml.base import config, BaseConfig
+from dffml.db.base import BaseDatabase, Condition
+from dffml.db.sqlite import SqliteDatabase
+from dffml.record import Record
+from dffml.source.source import BaseSource, BaseSourceContext
+from dffml.util.entrypoint import entrypoint
+
+
+@config
+class DbSourceConfig(BaseConfig):
+    db_implementation: str
+    db_config: any
+    table_name: str
+    model_columns: List[str]
+
+
+class DbSourceContext(BaseSourceContext):
+    async def update(self, record: Record):
+        model_columns = self.parent.config.model_columns.split()
+        key_value_pairs = collections.OrderedDict()
+        for key in model_columns:
+            if key.startswith("feature_"):
+                modified_key = key.replace("feature_", "")
+                key_value_pairs[key] = record.data.features[modified_key]
+            elif "_value" in key:
+                target = key.replace("_value", "")
+                if record.data.prediction:
+                    key_value_pairs[key] = record.data.prediction[target][
+                        "value"
+                    ]
+                else:
+                    key_value_pairs[key] = "undetermined"
+            elif "_confidence" in key:
+                target = key.replace("_confidence", "")
+                if record.data.prediction:
+                    key_value_pairs[key] = record.data.prediction[target][
+                        "confidence"
+                    ]
+                else:
+                    key_value_pairs[key] = 1
+            else:
+                key_value_pairs[key] = record.data.__dict__[key]
+        async with self.parent.db_impl() as db_ctx:
+            # Insert AND update since some tests don't insert before updating
+            # TODO: Maybe change the sqlite update query to use INSERT INTO ... ON DUPLICATE KEY UPDATE
+            # Note that this would impose all other db implementations to use "update or create if not exists" functionality
+            await db_ctx.insert(self.parent.config.table_name, key_value_pairs)
```

Comment:
Will this work?

https://github.com/intel/dffml/blob/cf4ee851414bef9dab33b5c037afefd6577932fa/dffml/db/sqlite.py#L73-L88